### PR TITLE
Changed from AWS basic authentication to DefaultAWSCredentialsProviderChain

### PR DESCRIPTION
**Pull request by shreepadangg**
Currently, Open source Kafka connector(https://github.com/instaclustr/kafka-connect-connectors) uses AWS credentials supplying arguments to connect AWS S3. 

We are modifying without supplying any arguments to find AWS credentials by using `DefaultAWSCredentialsProviderChain` 

Ref : https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,12 @@ target
 .classpath
 .java-version
 *.iml
+.DS_Store
```
Comment (zarakay) [inline .gitignore:8]: You only need this line, not the rest of it

File: s3/src/main/java/com/instaclustr/kafka/connect/s3/AwsStorageConnectorCommonConfig.java [hunk 2]
```diff
@@ -69,12 +63,12 @@ public static void verifyS3CredentialsAndBucketInfo(final Map<String, String> se
...
-                    addErrorMessageToConfigObject(configObject, AWS_ACCESS_KEY_ID, "The defined aws.accessKeyId is invalid");
-                    break;
-                case "SignatureDoesNotMatch":
-                    addErrorMessageToConfigObject(configObject, AWS_SECRET_KEY, "The defined aws.secretKey is invalid");
-                    break;
+	            case "InvalidAccessKeyId":
+	                addErrorMessageToConfigObject(configObject, "", "The AWS AccessKeyId is invalid");
+	                break;
+	            case "SignatureDoesNotMatch":
+	                addErrorMessageToConfigObject(configObject, "", "The AWS SecretKey is invalid");
```
Comment (zarakay) [inline s3/src/main/java/com/instaclustr/kafka/connect/s3/AwsStorageConnectorCommonConfig.java:70]: These error messages no longer make sense now that aws access key/secret are removed

**Review by zarakay — CHANGES_REQUESTED** [review]

File: s3/src/main/java/com/instaclustr/kafka/connect/s3/AwsStorageConnectorCommonConfig.java (hunk 2 above)
Comment (gi-el) [reply s3/src/main/java/com/instaclustr/kafka/connect/s3/AwsStorageConnectorCommonConfig.java:70]: The default chain still picks up the properties:
> Java System Properties - aws.accessKeyId and aws.secretKey

So I think we can even just keep it attached to the AWS_ACCESS_KEY_ID / AWS_SECRET_KEY.

This makes it double silly they went out of their way not to use the default chain...

File: .gitignore (hunk 1 above)
Comment (shreepadangg) [reply .gitignore:8]: Done

**Review by gi-el — APPROVED** [review]
I would personally keep the `AWS_SECRET_KEY` and `AWS_ACCESS_KEY_ID` constants and the error messages as is, however, it's fine like this also, because...
...(after discussion with Zara I've figured out that) once we're done with the Terraform overhaul we can use the Kafka Connect provider instead of manually instantiating the connector. If we do it from Terraform, we can create a specific AWS user for each connector and have Terraform pass the credentials. At that point, we wouldn't use the instance IAM role anymore, and we can just use the Instaclustr connector directly.

So to summarize: use this version now (install in bootscript, can even just pull from git and build on instance initialization), move to Instaclustr one later.

**Review by malateshs49 — APPROVED** [review]
